src/main/groovy/com/codebrig/arthur/observe/structure/filter/NameFilter.groovy

```diff
@@ -24,9 +24,27 @@ class NameFilter extends StructureFilter<NameFilter, String> {
                 return evaluateProperty(node.token)
             }
 
-            def childNameFilter
+            def childNameFilter = null
             if (node.language == SourceLanguage.Javascript) {
-                childNameFilter = new InternalRoleFilter("id").getFilteredNodes(node.children)
+                switch (Objects.requireNonNull(node).internalType) {
+                    case "FunctionDeclaration":
+                    case "FunctionExpression":
+                    case "ArrowFunctionExpression":
+                        childNameFilter = new InternalRoleFilter("id").getFilteredNodes(node.children)
+                        break
+                    case "ObjectMethod":
+                        childNameFilter = MultiFilter.matchAll(
+                                new TypeFilter("Identifier"), new InternalRoleFilter("key")
+                        ).getFilteredNodes(node.children)
+                        break
+                    case "NewExpression":
```

Comment:
I think the same deal here. This filter isn't meant to handle dynamically created functions. I created an issue to address it at some point in the future (#65).

Reply:
@BFergerson Acknowledged thanks, will commit changes later today.